Documentation/gettingstarted/encryption.rst

```diff
@@ -115,6 +116,31 @@ traffic) can be enabled as well. See :ref:`node_to_node_encryption` below.
 At this point the Cilium managed nodes will be using IPsec for all traffic. For further
 information on Cilium's transparent encryption, see :ref:`ebpf_datapath`.
 
+.. _encryption_ipam_eni:
+
+Encryption with IPAM ENI mode
+==============================
+
+In order to enable encryption on EKS with the IPAM ENI mode, deploy cilium via
+Helm with the following options:
+
+.. parsed-literal::
+
+    helm install cilium |CHART_RELEASE| \\
+      --namespace kube-system \\
+      --set ipam.mode=eni \\
+      --set tunnel=disabled \\
+      --set encryption.enabled=true \\
+      --set encryption.nodeEncryption=false
+
+By default, this enables routing via the ``cilium_host`` interface.
+Specifying encryption interfaces with ``--set encryption.interface=ethX`` is not supported.
+
+.. note::
+
+    For IPAM ENI mode, the user-provided interface may lead to issues where traffic
+    traffic may not get decrypted.
```

Comment:
Per Slack conversation, you said you were planning to file an issue to document the edge cases? I can link to the issue here.

Reply:
I agree, saying "this might not work" without providing any further details (why, how to identify the problem, how to fix) only serves to confuse users.

We should clearly define what we expect to happen. If there's a common pitfall, ideally we would mitigate it in software, but that's not possible, we should add a troubleshooting subsection that defines how you identify an issue like this and how you fix it.

Reply:
^ One way to tackle this is something like saying:

It is not required to specify `--interfaces` (or whatever the option is called). If however you do specify this flag, it will also configure the set of devices that decryption is applied on. As a result, if this option is misconfigured then it can cause connectivity issues between nodes as encrypted traffic cannot be decrypted at the destination. See the troubleshooting section (link here) for more details to identify and resolve this issue.